src/hotspot/share/opto/type.hpp

```diff
@@ -845,6 +849,39 @@ class TypeVectZ : public TypeVect {
   TypeVectZ(const Type* elem, uint length) : TypeVect(VectorZ, elem, length) {}
 };
 
+class TypeVMask : public Type {
+  const uint _elem_size;  // Element size in bytes of the masked vector
+  const uint _length;     // Number of elements in the masked vector
+public:
+  TypeVMask(uint elem_size, uint length) :
+    Type(VMask), _elem_size(elem_size), _length(length) {}
+  TypeVMask(const BasicType elem_bt, uint length) : Type(VMask),
+    _elem_size(type2aelembytes(elem_bt)), _length(length) {}
+
+  static const TypeVMask* make(const BasicType elem_bt, uint length) {
+    return (TypeVMask*)(new TypeVMask(elem_bt, length))->hashcons();
+  }
+
+  static const TypeVMask* make(uint elem_size, uint length) {
+    return (TypeVMask*)(new TypeVMask(elem_size, length))->hashcons();
+  }
+
+  static const TypeVMask* VMASK;
+
+  uint element_size_in_bytes(void) const {
+    return _elem_size;
+  }
+  uint length(void) const {
+    return _length;
+  }
+  virtual bool empty(void) const;
+  virtual bool eq(const Type* t) const;
+  virtual int hash() const;  // Type specific hashing
+  virtual bool singleton(void) const;
+  virtual const Type* xdual() const;
+  virtual const Type* xmeet(const Type* t) const;
```

Comment:
Extending TypeVector for creating new mask type seems like a logical extension, this has been done for AVX-512 predicated register support, it could be used for SVE also once the patch lands into mainline.

https://github.com/openjdk/jdk/pull/2768